docs/Hosting.md

```diff
@@ -0,0 +1,1181 @@
+
+Rules:
+* The safe registry MUST be derived programmatically at startup from tool metadata annotations (e.g., `safety=static`), not hard-coded lists duplicated in multiple places.
+* Removing a pattern or tightening its constraints MUST NOT silently broaden the existing safe surface (fail-safe principle).
+* A change proposal adding a new pattern MUST include a threat analysis and rollback plan (see 20.7).
+* Concrete tool identifiers MAY appear in implementation documentation (`safe-tools.md`) but remain informational; this section governs eligibility rules only.
+
+### 20.3 Azure Foundry Catalog Listing
+Foundry model and catalog listing endpoints MAY be exposed without authentication provided:
+1. Only public/global catalog entries (no tenant-scoped custom models) are returned.
+2. No subscription, deployment identifiers, or tenant IDs are included.
+3. Response is rate limited and size-limited (e.g., top N models) to mitigate enumeration abuse.
+4. Output includes a disclaimer: "No authenticated Azure context; data may be partial or cached." (spec language only; implementation detail separate).
+
+### 20.4 Exclusions (Not Safe Without Auth)
+| Category | Reason |
+|----------|--------|
+| Any command accepting `--subscription`, `--resource-group`, or resource IDs | Potential enumeration / access scope inference |
+| Storage / Data plane operations | Risk of surfacing tenant data paths even with read-only meta queries |
+| Key Vault, Secrets, Certificates, Keys | High sensitivity |
+| Service Bus, Cosmos DB, PostgreSQL, SQL, Redis | Potential metadata leakage |
+| Workbooks, Monitor (Logs/Metrics), Load Testing | Environment topology + performance characteristics |
+| Deployment / Create / Update / Delete commands | Mutative operations |
+
+### 20.5 Governance & Review
+1. Security review MUST sign off each candidate before activation in Demo Mode.
+2. A regression test suite SHOULD assert that unauthenticated activation of any excluded tool returns an authorization error.
+3. Telemetry SHOULD capture tool invocation counts to reassess risk and utility.
+4. A documented process MUST exist to remove a tool from the safe list if post-release risk emerges.
+
+### 20.6 Configuration Interaction
+* Demo Mode MUST automatically restrict tool registry to the enumerated safe set.
+* Attempting to invoke a non-safe tool while unauthenticated MUST yield a standardized error (e.g., `{"error":"auth_required"}`) without disclosing whether the tool exists outside safe mode.
+* Safe tools MUST tag responses with a context attribute (e.g., `"authMode":"none"`).
+
+### 20.7 Future Expansion Guardrails
+Any proposal to add more unauthenticated tools MUST provide: threat analysis, data classification, abuse scenarios, rate limit plan, rollback plan.
+
+### 20.8 Open Questions (Linked to Section 16)
+| Question | Link |
+|----------|------|
+| Should Foundry deployment listing remain excluded despite being non-mutative? | Section 16 (Demo Mode Scope) |
```

Comment:
Yes, I think foundry deployment listing should remain exclude. It requires authentication to Azure and would take tenant and project endpoint from Foundry as parameters.
Instead, we plan to add a new tool GetModelDetail - like ListModels, this uses read-only public API that doesn't require authentication.